compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1678,6 +1681,16 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
     fn encode_rendered_const_for_body(&mut self, body_id: hir::BodyId) -> String {
         let hir = self.tcx.hir();
         let body = hir.body(body_id);
+        if let ExprKind::Lit(lit) = &body.value.kind {
+            if let LitKind::Str(_, StrStyle::Included)
+            | LitKind::ByteStr(_, StrStyle::Included)
+            | LitKind::CStr(_, StrStyle::Included) = lit.node
+            {
+                // Literals from include_str!/include_bytes! are ignored by rustdoc anyway, so we
```

Comment:
How are they ignored?
I couldn't find any mentions of `include_*` macros in rustdoc.

Reply:
They get ignored via this logic in this function, because any slice literal is classified as Complex: https://github.com/rust-lang/rust/blob/59a8294849358a878a72358aa6d5fe5b9d312867/src/librustdoc/clean/utils.rs#L395

Reply:
This logic in rustdoc only deals with expressions belonging to the local crate (otherwise HIR could not be used), so it is not related to reading `rendered_const` from metadata.

Reply:
Also the printing logic is different between `print_const_expr` used for local constants, and `encode_rendered_const_for_body` used for external constants.

I think the logic from `print_const_expr` should be shared between rustc_metadata and rustdoc (moved to some place in `rustc_hir_pretty`?).
Then you won't even need preserving `StrStyle::Included` in AST - the rustdoc logic doesn't used, it throws away included bytes due to the `!value.span.from_expansion()` condition, most likely.

Reply:
It took me a while to figure out what you meant here, phew. I've never touched this part of the codebase before.

I think I figured out what you were pointing out. It's funny that this changed one of the rustdoc tests; I suppose we aren't very consistent about how literal suffixes are rendered.